### PR TITLE
Revise the Go quickstart

### LINKED ISSUE #159: Revise the Go quickstart
This is a tracking bug for the work required for https://github.com/census-instrumentation/opencensus-website/issues/158.

### PR DESCRIPTION
**Pull request by rakyll**
* Have more focus on the gRPC/HTTP integrations.
* Show how to use the low-level APIs for custom instrumentation.
* Mention exporters.
* Add link to the GitHub repo.

Fixes #159.

### CODE REVIEW

File: content/go.md [hunk 1]
```diff
@@ -39,109 +40,151 @@ $ go get go.opencensus.io
...
+
+OpenCensus stats collection happens in two stages:
+
+* Definition of measures and recording of data points
+* Definition of views and aggregation of the recorded data
+
+### Recording
+
+Measurements are data points associated with a measure.
+Recording implicitly tags the set of Measurements with the tags from the
```
Comment (songy23) [inline content/go.md]: This sentence is hard to read, how about "Recording the set of Measurements with implicit tags from the provided context"?

File: content/go.md [hunk 2]
```diff
@@ -39,109 +40,151 @@ $ go get go.opencensus.io
...
-  // Process the video.
-  process(ctx)
-
-  // Wait for a duration longer than reporting
-  duration to ensure the stats
-  // library reports the collected data.
-  fmt.Println("Wait longer than the reporting duration...")
-  time.Sleep(2 * time.Second)
+```go
+if err := view.Register(&view.View{
```
Comment (songy23) [inline content/go.md]: Add `[]tag.Key`?

File: content/go.md [hunk 3]
```diff
@@ -39,109 +40,151 @@ $ go get go.opencensus.io
...
+	Name:        "example.com/video_size_distribution",
+	Description: "processed video size over time",
+	Measure:     videoSize,
+	Aggregation: view.Distribution(0, 1<<32, 2<<32, 3<<32),
+}); err != nil {
+	log.Fatalf("Failed to register view: %v", err)
 }
+```
+
+Register begins collecting data for the view. Registered views' data will be
```
Comment (songy23) [inline content/go.md]: nit: s/begins collecting data/begins data collection

**Comment by rakyll** [comment]
All addressed, PTAL.

**Pull request merged**